core/templates/domain/exploration/StatesObjectFactorySpec.ts

```diff
@@ -386,6 +500,243 @@ describe('States object factory', () => {
             }
           },
         }));
+
+      expect(newStates.hasState('new state')).toBe(true);
+      expect(newStates.getStateNames()).toEqual(['first state', 'new state']);
+      expect((Object.keys(newStates.getStateObjects())).length).not.toBe(0);
+
+      newStates.setState('new state',
+        sof.createFromBackendDict('new state', {
+          classifier_model_id: null,
```

Comment:
I see that you are using this same dice in a lot of places. Can you use a global variable in that case, and not copy paste this everywhere?